activestorage/active.py

```diff
@@ -95,21 +95,22 @@ def __init__(self, uri, ncvar, storage_type=None, missing_value=None, _FillValue
         # FIXME: There is an outstanding issue with ._FilLValue to be handled.
         # If the user actually wrote the data with no fill value, or the
         # default fill value is in play, then this might go wrong.
-        if (missing_value, _FillValue, valid_min, valid_max) == (None, None, None, None):
-            if storage_type is None:
-                ds = Dataset(uri)
-            elif storage_type == "s3":
-                with load_from_s3(uri) as _ds:
-                    ds = _ds
-            try:
-                ds_var = ds[ncvar]
-            except IndexError as exc:
-                print(f"Dataset {ds} does not contain ncvar {ncvar!r}.")
-                raise exc
+        if storage_type is None:
+            ds = Dataset(uri)
+        elif storage_type == "s3":
+            with load_from_s3(uri) as _ds:
+                ds = _ds
+        try:
+            ds_var = ds[ncvar]
+        except IndexError as exc:
+            print(f"Dataset {ds} does not contain ncvar {ncvar!r}.")
+            raise exc
+
+        # FIXME: We do not get the correct byte order on the Zarr Array's dtype
+        # when using S3, so capture it here.
+        self._dtype = ds_var.dtype
```

Comment:
yes, always (even asked @davidhassell and @bnlawrence just to be 120% sure). BTW pls don't use `/` in branch names, bud - it confuses the heck out of my local `refs`